### ISSUE TITLE
Inconsistency of experiments

### ISSUE THREAD

**Issue by skylooop** [chongyi-zheng/infom#3]
Hi! Thank you for wonderful work.

I have question regarding FB: for some reason the baselines for pure FB reported in paper and when run locally do not match those reported in other papers. I.e exorl cheetah_run experiment reaches maximum 50 episode reward on FB (without IQL finetuning), while FB+IQL achieves max of 100, while in [1, 2] exorl cheetah_run on FB achieves 250-300 score.
[1] Touati et al. Does Zero-Shot Reinforcement Learning Exist
[2] Park et al. Foundation Policies with Hilbert Representations

Can you explain where inconsistency comes from?

**Comment by chongyi-zheng** [comment]
Thanks for the question! Various reasons may affect the performance, and I will mention two potential reasons here.

First, the FB implementation in [1, 2] used the official [implementation](https://github.com/facebookresearch/controllable_agent), while our implementation was adapted from [this repo](https://github.com/enjeeneer/zero-shot-rl). Because the second repo considers the offline RL problem, they added a bunch of implementation tricks to adapt to the offline setting, e.g., [half batch of the latent zs are from a normalized Gaussian distribution and half batch of the latent zs are from the backward representations](https://github.com/enjeeneer/zero-shot-rl/blob/2280cdf8fd8a5ccd51593ce9e6e06d27636c5a44/agents/fb/agent.py#L173-L185). 

Second, the hyperparameters used in the prior work are different from the hyperparameters used in this work, e.g., the batch size. Also, we update the actor and critic asynchronously: the `actor_freq`. Some hyperparameters may have a huge impact on the performance. This is the reason why we implemented the baselines from scratch to make fair comparisons.

Feel free to close the question.

**Comment by chongyi-zheng** [comment]
Thanks for the update! I initially included the additional cross multiplications for the Q in the actor loss to increase the batch size from $B$ to $B^2$. Since the policy also depends on the latent $z$, it makes sense to use the diagonal elements of the outer product of `forward_reprs` and `latents`. Did you find that the new `einsum` improves the performance as well?

To clarify the BC term, did you still consider the offline setting? Did you use the same hyperparameters and dataset size as in HILP?

Feel free to create a pull request, and I can try to merge the new code.

**Comment by skylooop** [comment]
I rerun my experiments and found that 1) the score with BC term slightly better than without it. 2) new `einsum` indeed performs much better. Cross multiplication version of `einsum` (as in your code) achieves performance x2 times worse, compared to taking `jnp.einsum('esd,sd->es', forward_reprs, latents)`. I will create pull request a bit later
